R/data.table.R

```diff
@@ -936,6 +936,42 @@ replace_dot_alias = function(e) {
         else
           x
       }
+      # extract the same codes from setDT() and do_j_names() for #6702
+      simple_extract <- function(name, x, checkSingleChar = TRUE, envir = parent.frame()) {
+        k = eval(name[[2L]], envir, envir)
+        if (is.list(k)) {
+          origj = j = if (name[[1L]] == "$") {
```

Comment:
try and match the original style of the code, it's much harder to read when the diff is larger than necessary